### PR TITLE
StudyArea creation form

### LINKED ISSUE #124: StudyArea form
Studies have attached [StudyAreas](https://github.com/datamade/django-pldp/blob/master/pldp/models/study.py#L8). We need a form to create these!

### PR DESCRIPTION
**Pull request by beamalsky**
## Overview

Closes #124.

This PR creates an interface to create new Study Areas and select them in the Study creation form. It also implements `crispy-forms` for the start of some prettier and cleaner forms. In making this I've come to the feeling that we should pre-populate some Philly-specific areas. See #129 

## To do

- [x] Write tests

## Testing Instructions

* Run migrations `python manage.py`
* Using the sidebar, navigate to the page to create a new study
* Create a new study location when you get to that point in the form. Is it intuitive? Did it work?

### CODE REVIEW

File: surveys/forms.py [hunk 1]
```diff
@@ -1,21 +1,57 @@
...
+        create_default_helper(self)
+
+    class Meta:
+        model = StudyArea
+        fields = '__all__'
+
+        leaflet_widget_attrs = {
+            'map_height': '400px',
+            'map_width': '100%',
+            'display_raw': 'true',
```
Comment (jeancochrane) [inline surveys/forms.py]: Does this option toggle the `Geojson` box? If so, what do you think about hiding it for the time being? It's a cool feature but it sort of overwhelms the page and I think it'll take a bit of work to get right. My instinct is to see whether this is something UCD wants and then invest the time in it if it is.

File: surveys/forms.py [hunk 2]
```diff
@@ -42,3 +82,12 @@ class Meta:
     def __init__(self, *args, form_entry, **kwargs):
         self.form_entry = SurveyFormEntry.objects.get(id=form_entry)
         super().__init__(*args, **kwargs)
+
+
+def create_default_helper(self):
```
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/forms.py]: Nope, just inexperience with that pattern! I gave it a shot and it seems to be working, but could you double check my work?

File: surveys/static/css/custom.css [hunk 3]
```diff
@@ -204,6 +204,11 @@ a.article:hover {
     CONTENT STYLE
 ----------------------------------------------------- */
 
+.asteriskField {
+  color: var(--orange-color);
```
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/static/css/custom.css]: At the top of this file! 

```
:root {
  --dark-green-color: #91cbb0;
  --off-white-color: #fafafa;
  --darker-green-color:  #203f31;
  --orange-color: #e94e0c;
}
```

This is a pattern @reginafcompton showed me. Thanks @reginafcompton!

File: surveys/static/css/images/spritesheet.svg [hunk 4]
```diff
@@ -0,0 +1,155 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/static/css/images/spritesheet.svg]: These are vestigial, you're right! Deleting.

File: surveys/templates/base.html [hunk 5]
```diff
@@ -1,23 +1,31 @@
...
+
+    <!-- JS -->
+    <script type="text/javascript" src="{% static 'js/jquery-3.3.1.min.js' %}"></script>
+    <script type="text/javascript" src="{% static 'js/bootstrap.min.js' %}"></script>
+    <script type="text/javascript" src="{% static 'js/datatables.min.js' %}"></script>
+    {% leaflet_js %}
+    <script type="text/javascript" src="{% static 'js/select2.min.js' %}"></script>
+
+    {% leaflet_js plugins="forms" %}
+    {% leaflet_css plugins="forms" %}
```
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/templates/base.html]: Good point, these can be limited to where they're needed. I'll do that refiguring here.

File: surveys/templates/base.html [hunk 6]
```diff
@@ -1,23 +1,31 @@
...
   <head>
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
-    <!-- <link rel="icon" href="{% static 'images/favicon.jpg' %}"> -->
-
-    <!-- JS -->
-    <script type="text/javascript" src="{% static 'js/jquery-3.3.1.min.js' %}"></script>
-    <script type="text/javascript" src="{% static 'js/bootstrap.min.js' %}"></script>
-    <script type="text/javascript" src="{% static 'js/datatables.min.js' %}"></script>
+    <link rel="icon" href="{% static 'images/favicon.png' %}">
```
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/templates/base.html]: Thanks 😎

Not permanent but I'm pleased with it for now!

File: surveys/views.py [hunk 7]
```diff
@@ -31,11 +32,26 @@ class LocationCreate(CreateView):
...
     form_class = StudyCreateForm
     model = Study
     template_name = "study_create.html"
-    success_url = '/'
+    success_url = reverse_lazy('surveys-list')
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        url = reverse_lazy('study-areas-create')
+        extra_help_text = ' Don\'t see the area you need? Create a new one <a href="{}" target="_blank">here</a>'.format(url)
```
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/views.py]: Yes nice idea!

File: surveys/forms.py (hunk 1 above)
Comment (beamalsky) [reply surveys/forms.py]: Sure! If it comes up we can let UCD know this is an option.

**Comment by beamalsky** [comment]
Thanks for these comments! I think it would be pretty useful to pre-populate this Study Area table a bit with Philly and its neighborhoods. That's reflected in #129 and would mean that this Study Areas field is never empty. What do you think?

File: surveys/forms.py [hunk 8]
```diff
@@ -7,31 +7,33 @@
...
+        super(JustSpacesForm, self).__init__(*args, **kwargs)
+        self.create_default_helper()
 
+    def create_default_helper(self):
+        self.helper = FormHelper()
+        self.helper.form_class = 'form-horizontal'
+        self.helper.label_class = 'col-lg-2'
+        self.helper.field_class = 'col-lg-8'
+        self.helper.form_method = 'post'
+        self.helper.add_input(Submit('submit', 'Submit'))
```
Comment (jeancochrane) [inline surveys/forms.py]: Looks great! :+1:

**Comment by jeancochrane** [comment]
Thanks for these quick fixes! Am I right to assume that #129 will also involve setting a default study area in order to fix the empty selectbox? If that's the plan then I'm fine waiting for #129 in order to fix the box display, otherwise I would vote to give it a null option like `----` to match the other selectboxes.

**Comment by beamalsky** [comment]
@jeancochrane is this ready to merge by you, now that that selectbox width issue is resolved?

File: surveys/forms.py (hunk 1 above)
Comment (jeancochrane) [reply surveys/forms.py]: @beamalsky Looks like this straggler got lost in the shuffle, but once `dislpay_raw` is set to `false` and the GeoJSON box doesn't show I consider this ready to merge!

Comment (beamalsky) [reply surveys/forms.py]: Ah thanks for the catch!

**Pull request merged**